src/com/lftechnology/java/training/sanish/application/component/EmployeeHelper.java

```diff
@@ -0,0 +1,416 @@
+            String whereCondition = "employeeId=?";
+            EmployeeDao employeeDao = new EmployeeDao();
+            while (updateRow == 0) {
+                message = "\n >>" + Constants.TYPE_ADDRESS_LABEL;
+                LOGGER.log(Level.INFO, message);
+                address = UserInput.getAlphabeticWords(inputScanner);
+                DateFormat dateFormat = new SimpleDateFormat(Constants.DB_DATE_FORMAT);
+                Date date = new Date();
+                SetStatement = "address=?, modifiedAt=?";
+                updateRow = employeeDao.update(whereCondition, SetStatement, address, dateFormat.format(date), employee.getEmployeeId());
+                if (updateRow == 0) {
+                    message = "\n" + Constants.ERROR_MSG_LABEL + Constants.FAIL_UPDATE_ADDRESS_MSG + Constants.TRY_AGAIN_MSG;
+                    LOGGER.log(Level.INFO, message);
+                    edit = UserInput.getString(inputScanner);
+                    if (!edit.equals(Constants.YES_OPTION) && !edit.equals(Constants.FULL_YES_OPTION)) {
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Change department of user
+     *
+     * @param inputScanner {@link Scanner}
+     * @param employee     {@link Employee}
+     * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+     */
+    public static void changeDepartment(Scanner inputScanner, Employee employee) {
+        String message =
+                "\n" + Constants.USER_DEPARTMENT_LABEL + employee.getDepartment() + "\n>>" + Constants.CHANGE_DEPARTMENT_CONFORM_MSG;
+        LOGGER.log(Level.INFO, message);
+        String edit = UserInput.getString(inputScanner);
+        if (edit.equals(Constants.YES_OPTION) || edit.equals(Constants.FULL_YES_OPTION)) {
+            int updateRow = 0;
+            String department;
+            String SetStatement;
+            String whereCondition = "employeeId=?";
+            EmployeeDao employeeDao = new EmployeeDao();
+            while (updateRow == 0) {
+                message = "\n >>" + Constants.TYPE_DEPARTMENT_LABEL;
+                LOGGER.log(Level.INFO, message);
+                department = UserInput.getAlphabeticWords(inputScanner);
+                DateFormat dateFormat = new SimpleDateFormat(Constants.DB_DATE_FORMAT);
+                Date date = new Date();
+                SetStatement = "department=?, modifiedAt=?";
+                updateRow = employeeDao.update(whereCondition, SetStatement, department, dateFormat.format(date), employee.getEmployeeId());
+                if (updateRow == 0) {
+                    message = "\n" + Constants.ERROR_MSG_LABEL + Constants.FAIL_UPDATE_DEPARTMENT_MSG + Constants.TRY_AGAIN_MSG;
+                    LOGGER.log(Level.INFO, message);
+                    edit = UserInput.getString(inputScanner);
+                    if (!edit.equals(Constants.YES_OPTION) && !edit.equals(Constants.FULL_YES_OPTION)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)